### DISCUSSION TITLE
CMake 3.13

### DISCUSSION THREAD

**Discussion by johndoe71rus** [diasurgical/DevilutionX#1226] [General]
is is necessary use CMake 3.13?
Many distros still use 3.10, ubuntu 18.04 LTS for example.

**Comment by Trihedraf** [comment]
This is where the change happened and they say why they did it. #1183

**Comment by julealgon** [comment]
Isn't CMake just a tool? Can't it be updated? 

Sorry if this is a noob question, but this discussion makes zero sense to me as a Windows user (was wondering what the difference is with Linux).

**Reply by qndel** [reply]
> Isn't CMake just a tool? Can't it be updated?
> 
> Sorry if this is a noob question, but this discussion makes zero sense to me as a Windows user (was wondering what the difference is with Linux).

it can but from my memories when I've tried to do it, it's a huge pain in the ass

**Reply by julealgon** [reply]
> it's a huge pain in the ass

Isn't that how things go on Linux usually?

**Reply by AJenbo** [reply]
Try updating CMake in MSVC I dare you :P

**Comment by AJenbo** [comment]
It's needed for properly supporting Windows 64bit. Currently you can lower it to 3.10, but we may start to use some of the features to clean up the build scripts.

I would recommend you update CMake: `sudo apt-add-repository 'deb https://apt.kitware.com/ubuntu/ bionic main'`

**Comment by johndoe71rus** [comment]
Since this is a discussion, we can continue.
I have a task to run devilutionX on lubuntu 18.04 x32. I can't build because cmake 3.10. Updating from a third party repository is a sensitive issue.
I have successfully built devilutionX on lubuntu 20.04 x64 following the instructions given [Compiling-from-Source](https://github.com/diasurgical/devilutionX/wiki/Compiling-from-Source) But this version does not run on target 18.04 x32
```
/lib/i386-linux-gnu/libm.so.6: version `GLIBC_2.29 'not found (required by ./devilutionx)
/lib/i386-linux-gnu/libc.so.6: version `GLIBC_2.28 'not found (required by ./devilutionx)
```


Can I build a version that will run on older OS?

**Reply by AJenbo** [reply]
I see 3 options for you:
- Use another installation with 18.04 and an updated CMake on it and build it there.
- Use debian:stretch-backports to build as we do.
- Use a prebuild artifact from CircleCI

**Reply by glebm** [reply]
> Updating from a third party repository is a sensitive issue.

The repo posted above is from the CMake authors

You can also simply compile CMake from source, it is no more difficult than compiling DevilutionX from source

If you trust Canonical over Kitware (the CMake org) you can use the sources from Ubuntu 20.04
